.github/workflows/cleanup-old-release.yml

```diff
@@ -0,0 +1,110 @@
+name: Close Old Release PRs
+
+on:
+  schedule:
+    - cron: '0 0 * * 1'  # Run at midnight every Monday
+
+  workflow_dispatch:
+    inputs:
+      pr_title_pattern:
+        description: 'Pattern to match PR titles (regex)'
+        required: true
+        default: '^chore\(weekly-'
+        type: string
+
+  workflow_call:
+    inputs:
+      pr_title_pattern:
+        description: 'Pattern to match PR titles (regex)'
+        required: true
+        default: '^chore\(weekly-'
+        type: string
+
+env:
+  USE_GITHUB_APP_TOKEN: true
+  DEFAULT_PR_TITLE_REGEX: '^chore\(release-' # default regex to match PR titles in the loki-release repo
+  PR_TITLE_REGEX: >-
+    ${{
+      github.event_name == 'workflow_call' && github.event.inputs.pr_title_pattern ||
+      github.event_name == 'workflow_dispatch' && github.event.inputs.pr_title_pattern ||
+      env.DEFAULT_PR_TITLE_REGEX
+    }}
+
+jobs:
+  cleanup-old-releases:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: read
+
+    steps:
+      - name: Checkout repository
```

Comment:
a few of these steps are duplicates of steps we already have defined in jsonnet and shared by other workflows. how would you feel about making this workflow generated from the jsonnet so we can share these and only have to make changes in one place?